### ISSUE TITLE
More fine-grained highlighting of parameters in function signatures

### ISSUE THREAD

**Issue by timhoffm** [sphinx-doc/sphinx#6417]
Parameters in function signatures are currently parsed to a single html element such as
`<em class="sig-param">myparam=4</em>`

It would be nice to split this down further into classed spans, so that we could highlight different parts of the parameters separately. For example:
~~~
<em class="sig-param">
<span class="sig-param-name">myparam</span>=<span class="sig-param-value">4</span>
</em>
~~~
(line breaks only for better readability. Actual code wound not have linebreaks as they would result in whitespace in the rendered HTML.)

Things to consider:
- This requires further splitting at the parser level.
- Maybe even add a class for `=`?
- Type annotations could be considered as well, as a separate span.

This is a follow-up on #4390/#6401 and #5868.

**Comment by tk0miya** [comment]
Will do.

**Comment by Thom1729** [comment]
Instead of `em`, `var` seems semantically appropriate.

**Comment by tk0miya** [comment]
I just made a draft PR #6990 to realize this feature. Please check it if you have time :-)

**Comment by tk0miya** [comment]
Now I merged #6990. It allows you to decorate function signatures via CSS more flexibly. Could you try it please? I don't think it still does not cover the complex case like #5868. So we need to improve the output more.
Thanks,

**Comment by timhoffm** [comment]
@tk0miya Thanks! This works for me.
